core/src/main/java/com/linecorp/armeria/common/util/Unwrappable.java

```diff
@@ -87,4 +87,36 @@ default <T> T as(Class<T> type) {
     default Unwrappable unwrap() {
         return this;
     }
+
+    /**
+     * Unwraps this object and returns the innermost object being decorated.
+     * If this {@link Unwrappable} is the innermost object, this method returns itself. For example:
+     * <pre>{@code
+     * class Foo implements Unwrappable {}
+     *
+     * class Bar<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Bar(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * class Qux<T extends Unwrappable> extends AbstractUnwrappable<T> {
+     *     Qux(T delegate) {
+     *         super(delegate);
+     *     }
+     * }
+     *
+     * final Foo foo = new Foo();
+     * foo.unwrapAll() == foo;
+     *
+     * final Bar<Foo> bar = new Bar<>(foo);
+     * bar.unwrapAll() == foo;
+     *
+     * final Qux<Bar<Foo>> qux = new Qux<>(bar);
+     * qux.unwrapAll() == foo;
```

Comment:
How about also adding `unwrap()`?
```suggestion
     * Qux<Bar<Foo>> qux = new Qux<>(bar);
     * assert qux.unwrap() == bar;
     * assert qux.unwrapAll() == foo;
```